controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -133,36 +131,50 @@ bool wfa_ca::reply(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx, const std::st
     return message_com::send_cmdu(sd, cmdu_tx);
 }
 
-bool wfa_ca::parse_params(std::unordered_map<std::string, std::string> &params,
+/**
+ * @brief: Parse the strings vector which containing parameter names, and values into unordered map
+ * which will contain the parameter name as a map key, and paramaeter value as map value.
+ * The function receives as an input unordered map of parameters that are mandatory, which will
+ * will include non-mandatory parameters as output.
+ * 
+ * @param[in] command_tokens: A vector containing commands parameters name and values.
+ * @param[in/out] params: An unordered map containing mandatory parameters on input, and parsed 
+ * command as output.
+ * @param[out] err_string: Contains an error description if the function fails.
+ * @return: true if successful, false if not.
+ */
+bool wfa_ca::parse_params(const std::vector<std::string> &command_tokens,
+                          std::unordered_map<std::string, std::string> &params,
                           std::string &err_string)
 {
-    // create copy of all mandatory param names
+    // create a copy of all mandatory param names
     std::list<std::string> mandatory_params;
     for (const auto &param_name : params) {
         mandatory_params.push_back(param_name.first);
     }
 
     size_t mandatory_params_cnt = 0;
-    while (!finished_tokens()) {
-        std::string param_name = get_next_token();
+    for (auto it = std::next(command_tokens.begin()); it != command_tokens.end(); it++) {
+        std::string param_name = *it;
```

Comment:
if I want it to be a reference I would do:
`auto& param name = *it;`

But yes, we generally want to use auto, but in some cases, you would prefer to explicitly write the type because it helps you to understand the code better.